opentmi_client/main.py

```diff
@@ -4,86 +4,183 @@
+
+  def subcmd_version_handler(self, args):
+    import pkg_resources  # part of setuptools
+    versions = pkg_resources.require("opentmi_client")
+    if self.args.verbose >= 1:
+        for v in versions:
+            print(v)
+    else:
+        print(versions[0].version)
+    return EXIT_CODE_SUCCESS
+
+  def subcmd_list_handler(self, args):
+    client = OpenTmiClient(host=args.host, port=args.port)
+    if args.testcases:
+        testcases = client.get_testcases()
+        if args.json:
+            print(json.dumps(testcases))    
+        for tc in testcases:
+            print(tc['tcid'])
+    elif args.campaigns:
+        campaigns = client.get_campaign_names()
+        if args.json:
+            print(campaigns)
+        else:
+            for campaign in campaigns:
+                print(campaign)
+
+  def subcmd_store_build(self, args):
+    raise NotImplementedError()
+
+  def subcmd_store_testcase(self, args):
+    raise NotImplementedError()
+
+  def subcmd_store_result(self, args):
+    raise NotImplementedError()
+
+def opentmiclient_main():
     """
-    parser = argparse.ArgumentParser()
-
-    parser.add_argument('--version',
-                      dest='version',
-                      default=False,
-                      action='store_true',
-                      help='Prints package version and exits')
-
-    parser.add_argument('--host',
-                      dest='host',
-                      default='localhost',
-                      help='OpenTMI host, default: localhost')
-
-    parser.add_argument('--json',
-                        dest='json',
-                        default=False,
-                        action='store_true',
-                        help='results as json')
-    parser.add_argument('-p', '--port',
-                      dest='port',
-                      type=int,
-                      default=3000,
-                      help='OpenTMI port')
-
-    parser.add_argument('--list',
-                      dest='list',
-                      action='store_true',
-                      default=False,
-                      help='List something')
-    parser.add_argument('--testcases',
-                      dest='testcases',
-                      default='',
-                      help='Testcases')
-    parser.add_argument('--campaigns',
-                      dest='campaigns',
-                      default='',
-                      help='Campaigns')
-
-    args = parser.parse_args()
-    return args
-
-
-def opentmi_client_main():
-    """! Function used to drive CLI (command line interface) application
-    @return Function exits with success-code
-    @details Function exits back to command line with ERRORLEVEL
+    Function used to drive CLI (command line interface) application.
+    Function exits back to command line with ERRORLEVEL
+    Returns:
+        Function exits with success-code
     """
-    retcode = 0
+    cli = OpentTMIClientCLI()
+    exit(cli.execute())
```

Comment:
Why is exit() used instead of sys.exit()? According to documentation, exit() is useful in interactive shell but should not be used in programs, even though it essentially does the same things as sys.exit() (raise a SystemExit)